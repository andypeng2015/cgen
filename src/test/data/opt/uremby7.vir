module foo

export function w $foo(w %x) {
@start:
  %x = urem.w %x, 7_w
  ret %x
}
