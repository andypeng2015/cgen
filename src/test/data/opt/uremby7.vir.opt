module foo

export function w $foo(w %x) {
@2:
  %4 = umulh.w %x, 0x24924925_w ; @8
  %5 = lsl.w %4, 0x3_w ; @9
  %6 = sub.w %5, %4 ; @10
  %7 = sub.w %x, %6 ; @11
  ret %7
}
