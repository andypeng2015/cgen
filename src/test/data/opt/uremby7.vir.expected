module foo

export function w $foo(w %x) {
@0:
  %3 = umulh.w %x, 0x24924925_w ; @5
  %2 = lsl.w %3, 0x3_w ; @4
  %1 = sub.w %2, %3 ; @3
  %0 = sub.w %x, %1 ; @2
  ret %0
}
